lib/generators/declare_schema/migration/migrator.rb

```diff
@@ -336,18 +342,17 @@ def change_table(model, current_table_name)
           new_table_name = model.table_name
 
           db_columns = model.connection.columns(current_table_name).index_by(&:name)
-          key_missing = db_columns[model._defined_primary_key].nil? && model._defined_primary_key.present?
-          if model._defined_primary_key.present?
-            db_columns.delete(model._defined_primary_key)
+          if (pk = model._declared_primary_key.presence)
+            key_was_in_db_columns = db_columns.delete(pk)
           end
 
           model_column_names = model.field_specs.keys.map(&:to_s)
           db_column_names = db_columns.keys.map(&:to_s)
 
           to_add = model_column_names - db_column_names
-          to_add += [model._defined_primary_key] if key_missing && model._defined_primary_key.present?
+          to_add += [pk] unless key_was_in_db_columns
           to_remove = db_column_names - model_column_names
-          to_remove -= [model._defined_primary_key.to_sym] if model._defined_primary_key.present?
+          to_remove -= [pk.to_sym] if pk    # TODO: The .to_sym here means this code is always a no-op, right? -Colin
```

Comment:
Can you see any way that the `-=` would ever do anything? It seems to me that the `to_remove` array must always be Strings.

Reply:
Yea it does look like the `to_sym` is always making this a no-op. Everything is `to_s`'d before coming to this line. I get that this line is trying to remove the primary key from the list of potential columns to remove. This might be a latent bug that we need to fix by changing this to `to_remove -= [pk]`, because right now as this sits, it's possible that you can remove the primary key.

Reply:
^ @jebentier Do you agree? I could change it to just `pk.to_s`...but apparently no one is missing that it doesn't work that way now... so maybe it never happens that the PK column name is in this list at all?

Reply:
Oh hey, if you scroll up you can see that `to_remove` started from `db_column_names`, which came from `db_columns`, which already had the `pk` removed. Untangling the rat's nest is paying off already. 🎉 
I'll get rid of this line.

Reply:
Awesome! 🎉